igvm/hypervisor.py

```diff
@@ -215,6 +190,40 @@ def check_vm(self, vm, offline):
                     .format(*hw_pair)
                 )
 
+        # HV in supported state?
+        if self.dataset_obj['state'] not in ['online', 'online_reserved']:
+            raise InvalidStateError(
+                'Hypervisor "{}" is not in online state ({}).'
+                .format(self.fqdn, self.dataset_obj['state'])
+            )
+
+        # Enough CPUs?
+        if vm.dataset_obj['num_cpu'] > self.dataset_obj['num_cpu']:
+            raise HypervisorError(
+                'Not enough CPUs. Destination Hypervisor has {0}, '
+                'but VM requires {1}.'
+                .format(self.dataset_obj['num_cpu'], vm.dataset_obj['num_cpu'])
+            )
+
+        # Proper VLAN?
+        if not self.get_vlan_network(vm.dataset_obj['intern_ip']):
+            raise HypervisorError(
+                'Hypervisor "{}" does not support route_network "{}".'
+                .format(self.fqdn, vm.route_network)
+            )
+
+        # Those checks below all require libvirt connection,
+        # so execute them last to avoid unnecessary overhead if possible.
+
+        # Enough memory?
+        free_mib = self.free_vm_memory()
+        if vm.dataset_obj['memory'] > free_mib:
```

Comment:
This one is arguable. I think it is safe for us but maybe this should still remain a hypervisor preference because people might want to allow memory ballooning. What is your thought on this @TuxPowered42

Reply:
lets wait for feedback from @TuxPowered42 but addresse this in another pr since the logic is the same as before, just in another place

Reply:
We don't allow for memory ballooning. We under-allocate VMs and give them their own swap.